File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        num_map = {}
        count = []
        res = []

        # Map value to frequency
        for n in nums:
            num_map[n] = 1 + num_map.get(n, 0)

        # List of [count, value]
        for key, value in num_map.items():
            count.append([value, key])

        # sort ascending by count
        count.sort()

        # pop highest count value in range k
        for i in range(k):
            res.append(count.pop()[1])

        return res

File: Data Structures & Algorithms/max-consecutive-ones/submission-4.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int res = -1;
        int counter = 0;
        for (int num : nums){
            if (num == 1){
                counter++;
            } else {
                counter = 0;
            }

            if (counter > res){
                res = counter;
            }
        }
        return res;
    }
};